packages/core/src/http-client/http-client.spec.ts

```diff
@@ -222,6 +234,40 @@ describe('generic http client', () => {
       );
     });
 
+    it('is possible to execute against a non principal propagation destination by only providing the subdomain (iss) instead of the whole jwt', async () => {
+      mockServiceBindings();
+      mockServiceToken();
+
+      const nocks = [
+        mockInstanceDestinationsCall(nock, [], 200, onlyIssuerServiceToken),
+        mockSubaccountDestinationsCall(
+          nock,
+          certificateMultipleResponse,
+          200,
+          onlyIssuerServiceToken
+        ),
+        mockSingleDestinationCall(
+          nock,
+          certificateSingleResponse,
+          200,
+          'ERNIE-UND-CERT',
+          wrapJwtInHeader(onlyIssuerServiceToken).headers
+        ),
+        nock(certificateSingleResponse.destinationConfiguration.URL)
+          .get(/.*/)
+          .reply(200, 'iss token used on the way')
+      ];
+
+      const response = await executeHttpRequest(
+        { destinationName: 'ERNIE-UND-CERT' },
+        { method: 'get' },
+        {},
+        { iss: onlyIssuerXsuaaUrl }
+      );
+      allMocksUsed(nocks);
```

Comment:
Consider renaming this function to `expectAllMocksUsed`. It isn't entirely clear that this contains assertions.